measure_optimize.py

```diff
@@ -2,180 +2,2412 @@
+                        "linear_solver": "mumps",
+                        "max_iter": 3000,
+                        # "halt_on_ampl_error": "yes",
+                        "bound_push": 1e-10,
+                        "warm_start_init_point": "yes",
+                        "warm_start_bound_push": 1e-10,
+                        "warm_start_bound_frac": 1e-10,
+                        "warm_start_slack_bound_frac": 1e-10,
+                        "warm_start_slack_bound_push": 1e-10,
+                        "warm_start_mult_bound_push": 1e-10,
+                    }
+                },
+            )
+
+        # check solver status and report
+        # if convergted to optimal solution and feasible
+        if (results.solver.status == SolverStatus.ok) and (
+            results.solver.termination_condition == TerminationCondition.optimal
+        ):
+            if self.optimize_print_level >= 2:  # print level is higher
+                print("The problem is solved optimal and feasible. ")
+        # if get infeasible status
+        elif results.solver.termination_condition == TerminationCondition.infeasible:
+            if self.optimize_print_level >= 1:  # print level is lower to warn
+                print("The problem is solved to infeasible point. ")
+        # if there is other solver status
+        else:
+            if self.optimize_print_level >= 1:
+                print("Not converged. Solver status:", results.solver.status)
+
+    def optimizer(
+        self,
+        budget_opt,
+        initial_solution,
+        mixed_integer=False,
+        obj=ObjectiveLib.A,
+        mix_obj=False,
+        alpha=1,
+        fixed_nlp=False,
+        fix=False,
+        upper_diagonal_only=False,
+        num_dynamic_t_name=None,
+        budget=100,
+        init_cov_y=None,
+        initial_fim=None,
+        dynamic_install_initial=None,
+        total_measure_initial=1,
+        static_dynamic_pair=None,
+        time_interval_all_dynamic=False,
+        total_manual_num_init=10,
+        cost_initial=100,
+        fim_diagonal_small_element=0,
+        print_level=0,
+    ):
         """
+        Initialize, formulate, and solve the MO problem.
+        This function includes two steps:
+        1) Create the optimization problem (Pyomo model)
+        2) Initialize the binary variables in the model with initial binary solutions
+        3) Initialize other variables in the model, computed by the binary variable values
 
-        :param FIM:
-        :return:
+        Arguments
+        ---------
+        :param budget_opt: budget
+        :param initial_solution: a dictionary, key: budget, value: initial solution pickle file name
+            this option stores the available initial solutions with their budgets
+        :param mixed_integer: boolean
+            not relaxing integer decisions
+        :param obj: Enum
+            "A" or "D" optimality, use trace or determinant of FIM
+        :param mix_obj: boolean
```

Comment:
If mix_obj is true, then obj should be ignored. I find it strange that mix_obj requires obj=D.

Reply:
yes. fixed this logic